feat: add global search command

DJKnaeckebrot opened the pull request:
I used the cmdk command overlay to create a search function for projects and services. It is globally available via ctrl/cmd + j 

![image](https://github.com/user-attachments/assets/d109ba57-d3d4-4a35-a345-64f170564769)

The component queries the projects and services via the api and redirects to the specific page. It can be used with keyboard or mouse :)

190km commented:
Nice that looks pretty, what about a little circle with the status of the service ?

Siumauricio commented:
Hey Looks great!, however when we have services with the same name it hover all of them

![Screenshot 2024-12-10 at 9 59 24 PM](https://github.com/user-attachments/assets/e49c1f4a-fb14-4a27-aaa0-6f1d1c31eb1c)

also the @190km suggestions sounds good, can you add some status to the right side?

Siumauricio commented on apps/dokploy/components/dashboard/search-command.tsx:
```diff
@@ -0,0 +1,170 @@
+            </CommandItem>
+            <CommandItem
+              onSelect={() => {
+                router.push("/dashboard/profile");
```
```suggestion
                router.push("/dashboard/settings/server");
```

Siumauricio replied:
```suggestion
                router.push("/dashboard/settings/profile");
```

Siumauricio commented on apps/dokploy/components/dashboard/search-command.tsx:
```diff
@@ -0,0 +1,170 @@
+            </CommandItem>
+            <CommandItem
+              onSelect={() => {
+                router.push("/dashboard/docker");
```
```suggestion
                router.push("/dashboard/docker");
```

Siumauricio commented on apps/dokploy/components/dashboard/search-command.tsx:
```diff
@@ -0,0 +1,170 @@
+              onSelect={() => {
+                router.push("/dashboard/monitoring");
+                setOpen(false);
+              }}
+            >
+              Monitoring
+            </CommandItem>
+            <CommandItem
+              onSelect={() => {
+                router.push("/dashboard/traefik");
+                setOpen(false);
+              }}
+            >
+              Traefik
+            </CommandItem>
+            <CommandItem
+              onSelect={() => {
+                router.push("/dashboard/docker");
+                setOpen(false);
+              }}
+            >
+              Docker
+            </CommandItem>
+            <CommandItem
+              onSelect={() => {
+                router.push("/dashboard/requests");
+                setOpen(false);
+              }}
+            >
+              Requests
+            </CommandItem>
+            <CommandItem
+              onSelect={() => {
+                router.push("/dashboard/profile");
+                setOpen(false);
+              }}
+            >
+              Settings
+            </CommandItem>
+          </CommandGroup>
```
Can you hide the following items when is the cloud version?

you can use this endpoint

```ts
const { data: isCloud, isLoading } = api.settings.isCloud.useQuery();
```

- Monitoring
- Traefik
- Docker
- Requests

DJKnaeckebrot commented:
> Nice that looks pretty, what about a little circle with the status of the service ?

Sure I can see what I can do :)



> Hey Looks great!, however when we have services with the same name it hover all of them
> 
> ![Screenshot 2024-12-10 at 9 59 24 PM](https://private-user-images.githubusercontent.com/47042324/394566308-e49c1f4a-fb14-4a27-aaa0-6f1d1c31eb1c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM4OTg2NzgsIm5iZiI6MTczMzg5ODM3OCwicGF0aCI6Ii80NzA0MjMyNC8zOTQ1NjYzMDgtZTQ5YzFmNGEtZmIxNC00YTI3LWFhYTAtNmYxZDFjMzFlYjFjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjExVDA2MjYxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTljN2M2Zjc1ODhmOTQyZDM3ZGE2MzI1YTBlY2UwMTI2NTUwMGUwNmI2OTcxNzQ4YjhmOGYxYzYyY2VhYThjMTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.8jBYvEFUP0_ESuK36B_OkDY1PHaQaJvjpDsHeAtaB1Y)
> 
> also the @190km suggestions sounds good, can you add some status to the right side?

How about making the name project / servicename ? Than it should be clear I guess ? I'll also make it hide the other stuff on cloud setuo :)

DJKnaeckebrot commented:
![image](https://github.com/user-attachments/assets/3dfa1c68-9ef2-4aa8-bf24-9eaf433812ad)

@190km something like this? 

@Siumauricio Hope that fixed the issue with the same service names as it now should be clear which project it is in :)

190km commented:
Looks amazing !

Siumauricio commented:
Looks awesome now! @DJKnaeckebrot, let me know when you ready to check the changes, Thanks!

DJKnaeckebrot commented:
I pushed the latest changes, so should be ready for review @Siumauricio 

I added the application.id as hidden text which seems to resolve the issue for now. If anyone ever finds a better solution/workaround would be awesome!

I also implemented the suggestion from @drudge to use the same badge style to keep it consistent! Thanks for that!

![image](https://github.com/user-attachments/assets/3dc7346b-4a6b-4c73-9499-aa11f73435c6)

Siumauricio reviewed (approved):
Thanks you for this!

DZsolt01 commented:
Awesome work! Maybe add a little timeout between searches, if I understood the code correctly it will run at every down so for a "test" word will call 4 times.

DJKnaeckebrot commented:
> Awesome work! Maybe add a little timeout between searches, if I understood the code correctly it will run at every down so for a "test" word will call 4 times.

Thank you! You are actually right, it will rerender on every key down. I am not too sure if it will retrigger the API call tho, it uses react-query to it should be catched. But I'll add a debounce on the setSearch so it will wait 200ms till it changes the value to cut down on potential rerenders. Thanks for the hint @DZsolt01 !

@Siumauricio I'll implement a patch for this and open a new PR :)
